compiler/rustc_query_system/src/query/caches.rs

```diff
@@ -120,24 +120,20 @@ where
 
     #[inline(always)]
     fn lookup(&self, key: &K) -> Option<(V, DepNodeIndex)> {
-        // FIXME: lock_shard_by_hash will use high bits which are usually zero in the index() passed
-        // here. This makes sharding essentially useless, always selecting the zero'th shard.
-        let lock = self.cache.lock_shard_by_hash(key.index() as u64);
+        let lock = self.cache.lock();
         if let Some(Some(value)) = lock.get(*key) { Some(*value) } else { None }
```

Comment:
`IndexVec` is contiguous, right? Did the old implementation make much sense? From a quick glance, it seems to me that if the indices were in fact high enough to shard we'd end up with largely empty `IndexVec`s in each shard

Reply:
Yes, that sounds plausible. In any case IIRC the sharding uses the high bits and we'd almost certainly run out of memory first.